### DISCUSSION TITLE
Geração multi-browsers

### DISCUSSION THREAD

**Discussion by RatoX** [frontendbr/forum#84] [Perguntas e Respostas]
Galera atualmente vários apps nativos estão implementando seus próprios renders para não deixar o usuário do aplicativo deles.

Alguns deles US o nw.io pra isso e outros eu não sei. Minha pergunta é como ces tão lidando com isso?  Testando apenas responsividade e acreditando nos crossbrowsers que os frameworks entregam?

**Comment by kazzkiq** [comment]
O iOS possui apenas dois motores de renderização web: WKWebView e UIWebView. Atualmente o desenvolvedor pode escolher qual utilizar em seu aplicativo (sendo que o WKWebView é o recomendado por ser mais moderno).

Como a Apple não permite o desenvolvimento de outros motores de renderização no seu sistema, o desenvolvedor só precisa se preocupar com estes dois (que usam a engine WebKit).

Já no caso do Android o buraco é mais em baixo. Até onde eu sei, o desenvolvimento de outras engines web é 100% liberado no sistema da Google. Isso significa que cada navegador e/ou aplicativo pode implementar a sua própria, ou usar uma que já exista no mercado (desde que ela tenha sido portada para o Android).

Mas mesmo com essa liberdade de escolha no Android, a maior parte dos browsers continua se baseando ou no WebKit, ou na Blink (Google/Chrome).

Tendo dito isso, se fosse para dar preferência para uma engine na hora de desenvolver um app, eu escolheria a WebKit. Além de ser a engine usada no iOS, também é muito parecida com a Blink do Chrome (pra não dizer idêntica em muitos aspectos, já que a Blink saiu de dentro do WebKit).

Focando na engine Webkit te permite desenvolver apps que funcionem em iOS, e que tenham maior chance de rodar nos Androids (onde você pode escolher qual engine usar).

Desconheço como andam as coisas no Windows Phone.
